### PR TITLE
Fix unittest pipeline

### PR DESCRIPTION
**Pull request by pjanotti**
Brings the latest from upstream. Tests: so far only `docker-compose run build`

### CODE REVIEW

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+using Datadog.Trace.Vendors.Serilog.Core;
 using Datadog.Trace.Vendors.Serilog.Events;
 using Datadog.Trace.Vendors.Serilog.Sinks.File;
 
 namespace Datadog.Trace.Logging
 {
     internal static class DatadogLogging
     {
-        private const string NixDefaultDirectory = "/var/log/signalfx/";
+        private const string NixDefaultDirectory = "/var/log/signalfx/dotnet";
```
Comment (rmfitzpatrick) [inline src/Datadog.Trace/Logging/DatadogLogging.cs:16]: Since the logs are already prefixed why is this necessary?

File: customer-samples/AutomaticTraceIdInjection/Log4NetExample/Log4NetExample.csproj [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Datadog.Trace" Version="1.8.0" />
```
Comment (rmfitzpatrick) [inline customer-samples/AutomaticTraceIdInjection/Log4NetExample/Log4NetExample.csproj:9]: I think we'll need to ensure this is updated w/ SetAllVersions

File: customer-samples/AutomaticTraceIdInjection/Log4NetExample/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "profiles": {
+    "Log4NetExample": {
+      "commandName": "Project",
+      "environmentVariables": {
+        "DD_LOGS_INJECTION": "true"
```
Comment (rmfitzpatrick) [inline customer-samples/AutomaticTraceIdInjection/Log4NetExample/Properties/launchSettings.json:6]: All DD_ env vars should be updated to SIGNALFX_

File: customer-samples/AutomaticTraceIdInjection/NLog45Example/NLog45Example.csproj [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Datadog.Trace" Version="1.8.0" />
```
Comment (rmfitzpatrick) [inline customer-samples/AutomaticTraceIdInjection/NLog45Example/NLog45Example.csproj:9]: same version comment

File: reproductions/StackExchange.Redis.AssemblyConflict.SdkProject/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "profiles": {
+    "StackExchange.Redis.AssemblyConflict.SdkProject": {
+      "commandName": "Project",
+      "nativeDebugging": true,
+      "environmentVariables": {
+        "COR_ENABLE_PROFILING": "1",
+        "COR_PROFILER": "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}",
```
Comment (rmfitzpatrick) [inline reproductions/StackExchange.Redis.AssemblyConflict.SdkProject/Properties/launchSettings.json:8]: All app and launch settings for samples and reproductions need to be update to use our project id and SIGNALFX_ env vars

File: src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs [hunk 6]
```diff
@@ -38,6 +38,19 @@ public static Scope CreateOutboundHttpScope(Tracer tracer, string httpMethod, Ur
 
             try
             {
+                Span parent = tracer.ActiveScope?.Span;
+
+                if (parent != null &&
+                    parent.Type == SpanTypes.Http &&
```
Comment (rmfitzpatrick) [inline src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs:44]: CreateOutboundHttpScope isn't setting this type, but SpanKinds.Client Tags.SpanKind tag instead.

File: src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs [hunk 7]
```diff
@@ -47,11 +60,11 @@ public static Scope CreateOutboundHttpScope(Tracer tracer, string httpMethod, Ur
                 // Only the span responsible for propagated context should have client span.kind
                 if (span.Context.Parent != null)
                 {
-                    var parent = ((SpanContext)span.Context.Parent).Span;
-                    var spanKind = parent.GetTag(Tags.SpanKind);
+                    var spanParent = ((SpanContext)span.Context.Parent).Span;
```
Comment (rmfitzpatrick) [inline src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs:63]: nit, parentSpan for readability and convention

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs [hunk 8]
```diff
@@ -59,8 +66,8 @@ public void HttpClient_TracingDisabled()
             int agentPort = TcpPortProvider.GetOpenPort();
             int httpPort = TcpPortProvider.GetOpenPort();
 
-            using (var agent = new MockZipkinCollector(agentPort))
-            using (ProcessResult processResult = RunSampleAndWaitForExit(agent.Port, arguments: $"HttpClient TracingDisabled Port={httpPort}", envVars: ZipkinEnvVars))
+            using (var agent = new MockTracerAgent(agentPort))
```
Comment (rmfitzpatrick) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs:69]: Using the MockZipkinCollector and ZipkinEnvVars is how we continually verify our updates.

File: src/Datadog.Trace/Logging/DatadogLogging.cs (hunk 1 above)
Comment (pjanotti) [reply src/Datadog.Trace/Logging/DatadogLogging.cs:16]: Just following with symmetrical change from upstream.

File: src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs (hunk 6 above)
Comment (pjanotti) [reply src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs:44]: Is it guaranteed that the tag is going to be present? I mean if it returns null it will crash with NullReferenceException. For now, I'm going to invert, ie.: have it with the constants calling equal on the tag value (will change it below if appropriate too).

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs (hunk 8 above)
Comment (pjanotti) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs:69]: I assume that `MockTracerAgent` is kept for instrumentations not touched in this repo yet.

Comment (rmfitzpatrick) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs:69]: Overall yes, though we would potentially change their content indirectly through helper updates and haven't gone through the full evaluation.

File: src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs (hunk 6 above)
Comment (rmfitzpatrick) [reply src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs:44]: No, it will only exist if it's potentially part of a distributed request (injects headers and/or is associated with a client operation).
